packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/SVGImage/SVGImage.js

```diff
@@ -0,0 +1,54 @@
+import { PropTypes } from 'prop-types'
+import React, { useEffect } from 'react'
+import styled from 'styled-components'
+
+import { draggable } from '@plugins/drawingTools/components'
+
+export const DraggableImage = styled(draggable('image'))`
+  cursor: move;
+`
+
+const INVERT =
+  `<svg style="position: fixed; right: 100%; top: 100%; visibility: hidden;">
+    <defs>
+      <filter id="svg-invert-filter" color-interpolation-filters="sRGB">
+        <feComponentTransfer>
+          <feFuncR type="table" tableValues="1 0"/>
+          <feFuncG type="table" tableValues="1 0"/>
+          <feFuncB type="table" tableValues="1 0"/>
+        </feComponentTransfer>
+      </filter>
+    </defs>
+  </svg>`
+
+export default function SVGImage ({
+  invert,
+  move,
+  naturalHeight,
+  naturalWidth,
+  onDrag,
+  src,
+  subjectID
+}) {
+  useEffect(function onInvert () {
+    if (invert) {
+      if (!document.getElementById('svg-invert-filter')) {
+        document.body.insertAdjacentHTML('afterbegin', INVERT)
+      }
+    }
+  }, [invert])
```

Comment:
```suggestion
if (invert) {
  if (!document.getElementById('svg-invert-filter')) {
    document.body.insertAdjacentHTML('afterbegin', INVERT)
  }
}
```

The effect hook here [adds an extra render](https://beta.reactjs.org/apis/usestate#storing-information-from-previous-renders) that you probably don't need. Do you really need to check `invert` every time the button is toggled? It seems, to me, like you should add the SVG filter to the document on the first render, then skip this block on subsequent renders if the filter already exists.